### PR TITLE
feat/DEVSU-2389 update groups detailed view

### PR DESCRIPTION
**Pull request by bnguyen-bcgsc**
- DEVSU-2389
- Update detailed view for Groups to display users by their full name instead of object id

### CODE REVIEW

File: app/components/DataTable/components/DetailDialog/index.tsx [hunk 1]
```diff
@@ -89,7 +89,7 @@ const ObjectAttributes = ({
               expandIcon={<ExpandMoreIcon />}
             >
               <Typography variant="subtitle2">
-                {`${idx}: ${mappedValIsArray ? '[' : '{'}`}
+                {`${!!mappedVal?.username ? `${mappedVal?.firstName} ${mappedVal?.lastName}` : idx}: ${mappedValIsArray ? '[' : '{'}`}
```
Comment (elewis2) [inline app/components/DataTable/components/DetailDialog/index.tsx:92]: I take it this isn't something we can customize for individual tables?

Comment (elewis2) [reply app/components/DataTable/components/DetailDialog/index.tsx:92]: Could we add a second default - if 'name' is present, then display name?

Comment (bnguyen-bcgsc) [reply app/components/DataTable/components/DetailDialog/index.tsx]: Changes pushed

File: app/components/DataTable/components/DetailDialog/index.tsx [hunk 2]
```diff
@@ -89,7 +89,9 @@ const ObjectAttributes = ({
               expandIcon={<ExpandMoreIcon />}
             >
               <Typography variant="subtitle2">
-                {`${idx}: ${mappedValIsArray ? '[' : '{'}`}
+                {!!mappedVal?.name ? `${mappedVal.name}` : ''}
+                {!!mappedVal?.username ? `${mappedVal?.firstName} ${mappedVal?.lastName}` : ''}
+                {`${!mappedVal?.username && !mappedVal?.name ? idx : ''}: ${mappedValIsArray ? '[' : '{'}`}
```
Comment (kttkjl) [inline app/components/DataTable/components/DetailDialog/index.tsx:94]: this is supposed to be a generic component, adding all of these defeats the purpose

**Review by kttkjl — CHANGES_REQUESTED** [review]

File: app/components/DataTable/components/DetailDialog/index.tsx (hunk 2 above)
Comment (bnguyen-bcgsc) [reply app/components/DataTable/components/DetailDialog/index.tsx:94]: Can I make them optional props?

**Review by kttkjl — APPROVED** [review]

**Pull request merged**